### PR TITLE
 Add implementation of `otPlatRadioSetMacFrameCounterIfLarger`

### PR DESCRIPTION
**Pull request by canisLupus1313**
Adding an Openthread radio API `otPlatRadioSetMacFrameCounterIfLarger` implementation and the corresponding call to the IEEEE802154 driver.

### CODE REVIEW

File: include/zephyr/net/ieee802154_radio.h [hunk 1]
```diff
@@ -169,6 +169,12 @@ enum ieee802154_config_type {
 	/** Sets the current MAC frame counter value for radios supporting transmit security. */
 	IEEE802154_CONFIG_FRAME_COUNTER,
 
+	/** Sets the current MAC frame counter value if current counter is greater than passed value
```
Comment (jciupis) [inline include/zephyr/net/ieee802154_radio.h:172]:
```suggestion
	/** Sets the current MAC frame counter value if the provided value is greater than the current value
```

File: modules/openthread/platform/radio.c [hunk 2]
```diff
@@ -1145,6 +1145,15 @@ void otPlatRadioSetMacFrameCounter(otInstance *aInstance,
 	(void)radio_api->configure(radio_dev, IEEE802154_CONFIG_FRAME_COUNTER,
 				   &config);
 }
+
+void otPlatRadioSetMacFrameCounterIfLarger(otInstance *aInstance, uint32_t aMacFrameCounter)
+{
+    ARG_UNUSED(aInstance);
```
Comment (jciupis) [inline modules/openthread/platform/radio.c:1151]: nit: indent

Comment (canisLupus1313) [reply modules/openthread/platform/radio.c]: fixed

File: include/zephyr/net/ieee802154_radio.h (hunk 1 above)
Comment (canisLupus1313) [reply include/zephyr/net/ieee802154_radio.h]: I reworded it slightly differently than suggested.

**Review by jciupis — DISMISSED** [review]

**Review by mariuszpos — DISMISSED** [review]

**Review by maciejbaczmanski — DISMISSED** [review]

File: modules/openthread/platform/radio.c [hunk 3]
```diff
@@ -1145,6 +1145,15 @@ void otPlatRadioSetMacFrameCounter(otInstance *aInstance,
...
 				   &config);
 }
+
+void otPlatRadioSetMacFrameCounterIfLarger(otInstance *aInstance, uint32_t aMacFrameCounter)
+{
+	ARG_UNUSED(aInstance);
+
+	struct ieee802154_config config = { .frame_counter = aMacFrameCounter };
+	(void)radio_api->configure(radio_dev, IEEE802154_CONFIG_FRAME_COUNTER_IF_LARGER,
+					&config);
```
Comment (pdgendt) [inline modules/openthread/platform/radio.c:1155]: Nit: fix indent
```suggestion
	(void)radio_api->configure(radio_dev, IEEE802154_CONFIG_FRAME_COUNTER_IF_LARGER,
				   &config);
```

Comment (canisLupus1313) [reply modules/openthread/platform/radio.c]: fixed

**Review by jciupis — DISMISSED** [review]

**Review by maciejbaczmanski — DISMISSED** [review]

**Review by pdgendt — DISMISSED** [review]

File: west.yml [hunk 4]
```diff
@@ -83,7 +83,7 @@ manifest:
       groups:
         - hal
     - name: hal_nordic
-      revision: 6c9f23498ee6a225fee049c8c5b2e1ac760bd4a8
+      revision: pull/126/head
```
Comment (rlubos) [inline west.yml:86]: Please update SHA: `609d4b41fe9773cd294eb8656d99eb9385389e52`

Comment (canisLupus1313) [reply west.yml]: updated

**Review by rlubos — APPROVED** [review]

**Review by maciejbaczmanski — APPROVED** [review]

**Review by pdgendt — APPROVED** [review]

**Pull request merged**